zebra/interface.c

```diff
@@ -2083,6 +2087,22 @@ static void zebra_if_dplane_ifp_handling(struct zebra_dplane_ctx *ctx)
 					   name, ifp->ifindex, zif_slave_type, master_ifindex,
 					   (unsigned long long)flags);
 
+			/*
+			 * Interface promiscuity changes trigger spurious routing updates on HBN
+			 * uplink interfaces, causing route flushes and traffic disruption.
+			 *
+			 * Check if only promiscuity flag changed (from netlink change mask)
+			 */
+			if (promiscuity && (change_flags == IFF_PROMISC)) {
```

Comment:
what about the negative/disable case - this logic only runs when promisc is `true` ? Maybe you want to say "if change_flags == PROMISC, only update the flags and return" ?